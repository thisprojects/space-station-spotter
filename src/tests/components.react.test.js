import React from 'react';
import { 
  Errors,
  OverLay,
  GeoCode,
  MapResults,
  DisplayResults,
  LocationInput,
  SubmitButton,
  ChooseAnotherLocation,
} from '../components/components.js';
import renderer from 'react-test-renderer';
import {
  mockedResultsArray,
} from './mocked-results.js';

test('Errors Matches Snapshot', () => {
  const component = renderer.create(
    <Errors />,
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Overlay Matches Snapshot', () => {
  const component = renderer.create(
    <OverLay />,
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('GeoCode Matches Snapshot', () => {
  const component = renderer.create(
    <GeoCode />,
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('MapResults Matches Snapshot', () => {
  const component = renderer.create(
    <MapResults results={mockedResultsArray}/>,
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('DisplayResults Matches Snapshot', () => {
  const component = renderer.create(
    <DisplayResults results={mockedResultsArray}/>,
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Location Input Matches Snapshot', () => {
  const component = renderer.create(
    <LocationInput />,
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Submit Button Matches Snapshot', () => {
  const component = renderer.create(
    <SubmitButton />,
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Choose Another Location Matches Snapshot', () => {
  const component = renderer.create(
    <ChooseAnotherLocation />,
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});