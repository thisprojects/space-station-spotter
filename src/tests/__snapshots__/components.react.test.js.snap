// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Choose Another Location Matches Snapshot 1`] = `
<div
  className="choose-another-location"
>
  <button
    onClick={[Function]}
  >
    Choose Another Location
  </button>
</div>
`;

exports[`DisplayResults Matches Snapshot 1`] = `
<div
  className="results"
>
  <div
    className="result-row"
  >
    <p>
      1
      <span>
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-clock fa-w-16 "
          data-icon="clock"
          data-prefix="fas"
          focusable="false"
          role="img"
          style={Object {}}
          viewBox="0 0 512 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8zm57.1 350.1L224.9 294c-3.1-2.3-4.9-5.9-4.9-9.7V116c0-6.6 5.4-12 12-12h48c6.6 0 12 5.4 12 12v137.7l63.5 46.2c5.4 3.9 6.5 11.4 2.6 16.8l-28.2 38.8c-3.9 5.3-11.4 6.5-16.8 2.6z"
            fill="currentColor"
            style={Object {}}
          />
        </svg>
      </span>
       27 / Jan / 2020  at  15:3:8
    </p>
  </div>
  <div
    className="result-row"
  >
    <p>
      2
      <span>
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-clock fa-w-16 "
          data-icon="clock"
          data-prefix="fas"
          focusable="false"
          role="img"
          style={Object {}}
          viewBox="0 0 512 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8zm57.1 350.1L224.9 294c-3.1-2.3-4.9-5.9-4.9-9.7V116c0-6.6 5.4-12 12-12h48c6.6 0 12 5.4 12 12v137.7l63.5 46.2c5.4 3.9 6.5 11.4 2.6 16.8l-28.2 38.8c-3.9 5.3-11.4 6.5-16.8 2.6z"
            fill="currentColor"
            style={Object {}}
          />
        </svg>
      </span>
       27 / Jan / 2020  at  15:13:2
    </p>
  </div>
  <div
    className="result-row"
  >
    <p>
      3
      <span>
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-clock fa-w-16 "
          data-icon="clock"
          data-prefix="fas"
          focusable="false"
          role="img"
          style={Object {}}
          viewBox="0 0 512 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8zm57.1 350.1L224.9 294c-3.1-2.3-4.9-5.9-4.9-9.7V116c0-6.6 5.4-12 12-12h48c6.6 0 12 5.4 12 12v137.7l63.5 46.2c5.4 3.9 6.5 11.4 2.6 16.8l-28.2 38.8c-3.9 5.3-11.4 6.5-16.8 2.6z"
            fill="currentColor"
            style={Object {}}
          />
        </svg>
      </span>
       27 / Jan / 2020  at  15:8:57
    </p>
  </div>
  <div
    className="choose-another-location"
  >
    <button
      onClick={[Function]}
    >
      Choose Another Location
    </button>
  </div>
</div>
`;

exports[`Errors Matches Snapshot 1`] = `
<div
  className="location-error"
/>
`;

exports[`GeoCode Matches Snapshot 1`] = `
<div
  className="geocode"
>
  <button
    onClick={[Function]}
  >
    Use Current Location
  </button>
</div>
`;

exports[`Location Input Matches Snapshot 1`] = `
<div
  className="location-input-wrapper"
>
  <input
    id="location-input"
    placeholder="Choose Location"
    type="text"
  />
</div>
`;

exports[`MapResults Matches Snapshot 1`] = `
Array [
  <div
    className="result-row"
  >
    <p>
      1
      <span>
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-clock fa-w-16 "
          data-icon="clock"
          data-prefix="fas"
          focusable="false"
          role="img"
          style={Object {}}
          viewBox="0 0 512 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8zm57.1 350.1L224.9 294c-3.1-2.3-4.9-5.9-4.9-9.7V116c0-6.6 5.4-12 12-12h48c6.6 0 12 5.4 12 12v137.7l63.5 46.2c5.4 3.9 6.5 11.4 2.6 16.8l-28.2 38.8c-3.9 5.3-11.4 6.5-16.8 2.6z"
            fill="currentColor"
            style={Object {}}
          />
        </svg>
      </span>
       27 / Jan / 2020  at  15:3:8
    </p>
  </div>,
  <div
    className="result-row"
  >
    <p>
      2
      <span>
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-clock fa-w-16 "
          data-icon="clock"
          data-prefix="fas"
          focusable="false"
          role="img"
          style={Object {}}
          viewBox="0 0 512 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8zm57.1 350.1L224.9 294c-3.1-2.3-4.9-5.9-4.9-9.7V116c0-6.6 5.4-12 12-12h48c6.6 0 12 5.4 12 12v137.7l63.5 46.2c5.4 3.9 6.5 11.4 2.6 16.8l-28.2 38.8c-3.9 5.3-11.4 6.5-16.8 2.6z"
            fill="currentColor"
            style={Object {}}
          />
        </svg>
      </span>
       27 / Jan / 2020  at  15:13:2
    </p>
  </div>,
  <div
    className="result-row"
  >
    <p>
      3
      <span>
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-clock fa-w-16 "
          data-icon="clock"
          data-prefix="fas"
          focusable="false"
          role="img"
          style={Object {}}
          viewBox="0 0 512 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8zm57.1 350.1L224.9 294c-3.1-2.3-4.9-5.9-4.9-9.7V116c0-6.6 5.4-12 12-12h48c6.6 0 12 5.4 12 12v137.7l63.5 46.2c5.4 3.9 6.5 11.4 2.6 16.8l-28.2 38.8c-3.9 5.3-11.4 6.5-16.8 2.6z"
            fill="currentColor"
            style={Object {}}
          />
        </svg>
      </span>
       27 / Jan / 2020  at  15:8:57
    </p>
  </div>,
]
`;

exports[`Overlay Matches Snapshot 1`] = `
<div
  className="overlay"
  style={
    Object {
      "display": "none",
    }
  }
>
  <h1>
    Space Station Spotter
  </h1>
  <span
    className="icon"
  >
    <svg
      aria-hidden="true"
      className="svg-inline--fa fa-satellite-dish fa-w-16 fa-3x "
      data-icon="satellite-dish"
      data-prefix="fas"
      focusable="false"
      role="img"
      style={Object {}}
      viewBox="0 0 512 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M305.44954,462.59c7.39157,7.29792,6.18829,20.09661-3.00038,25.00356-77.713,41.80281-176.72559,29.9105-242.34331-35.7082C-5.49624,386.28227-17.404,287.362,24.41381,209.554c4.89125-9.095,17.68975-10.29834,25.00318-3.00043L166.22872,323.36708l27.39411-27.39452c-.68759-2.60974-1.594-5.00071-1.594-7.81361a32.00407,32.00407,0,1,1,32.00407,32.00455c-2.79723,0-5.20378-.89075-7.79786-1.594l-27.40974,27.41015ZM511.9758,303.06732a16.10336,16.10336,0,0,1-16.002,17.00242H463.86031a15.96956,15.96956,0,0,1-15.89265-15.00213C440.46671,175.5492,336.45348,70.53427,207.03078,63.53328a15.84486,15.84486,0,0,1-15.00191-15.90852V16.02652A16.09389,16.09389,0,0,1,209.031.02425C372.25491,8.61922,503.47472,139.841,511.9758,303.06732Zm-96.01221-.29692a16.21093,16.21093,0,0,1-16.11142,17.29934H367.645a16.06862,16.06862,0,0,1-15.89265-14.70522c-6.90712-77.01094-68.118-138.91037-144.92467-145.22376a15.94,15.94,0,0,1-14.79876-15.89289V112.13393a16.134,16.134,0,0,1,17.29908-16.096C319.45132,104.5391,407.55627,192.64538,415.96359,302.7704Z"
        fill="currentColor"
        style={Object {}}
      />
    </svg>
  </span>
  <div
    className="geocode"
  >
    <button
      onClick={[Function]}
    >
      Use Current Location
    </button>
  </div>
  <h2>
    Or
  </h2>
  <span
    className="controls"
  >
    <div
      className="location-input-wrapper"
    >
      <input
        id="location-input"
        placeholder="Choose Location"
        type="text"
      />
    </div>
    <div
      className="submit-button"
    >
      <button
        onClick={[Function]}
      >
        Submit
      </button>
    </div>
  </span>
  <div
    className="location-error"
  />
  <div
    className="loading toggle-hide"
  >
    <svg
      aria-hidden="true"
      className="svg-inline--fa fa-spinner fa-w-16 fa-spin fa-3x "
      data-icon="spinner"
      data-prefix="fas"
      focusable="false"
      role="img"
      style={Object {}}
      viewBox="0 0 512 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M304 48c0 26.51-21.49 48-48 48s-48-21.49-48-48 21.49-48 48-48 48 21.49 48 48zm-48 368c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48-21.49-48-48-48zm208-208c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48-21.49-48-48-48zM96 256c0-26.51-21.49-48-48-48S0 229.49 0 256s21.49 48 48 48 48-21.49 48-48zm12.922 99.078c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48c0-26.509-21.491-48-48-48zm294.156 0c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48c0-26.509-21.49-48-48-48zM108.922 60.922c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48-21.491-48-48-48z"
        fill="currentColor"
        style={Object {}}
      />
    </svg>
  </div>
</div>
`;

exports[`Submit Button Matches Snapshot 1`] = `
<div
  className="submit-button"
>
  <button
    onClick={[Function]}
  >
    Submit
  </button>
</div>
`;
